src/equations/acoustic_perturbation_2d.jl

```diff
@@ -0,0 +1,163 @@
+@doc raw"""
+    AcousticPerturbationEquations2D(v_mean, rho_mean, c_mean)
+
+!!! warning "Experimental code"
+    This system of equations is experimental and may change in any future release.
+
+Acoustic perturbation equations in two space dimensions with constant mean flow. `v_mean` is a tuple
+of size two holding the mean velocities in each direction. `rho_mean` and `c_mean` are scalars
+representing the mean density and mean speed of sound, respectively.
+
+The equations are based on the APE-4 system introduced in the following paper:
+
+R. Ewert, W. Schröder
+"Acoustic perturbation equations based on flow decomposition via source filtering",
+Journal of Computational Physics,
+Volume 188, Issue 2,
+2003,
+[DOI: 10.1016/S0021-9991(03)00168-2](https://doi.org/10.1016/S0021-9991(03)00168-2).
+"""
+struct AcousticPerturbationEquations2D{RealT<:Real} <: AbstractAcousticPerturbationEquations{2, 3}
+  v_mean::SVector{2, RealT}
+  rho_mean::RealT
+  c_mean::RealT
+end
+
+function AcousticPerturbationEquations2D(v_mean::NTuple{2,<:Real}, rho_mean::Real, c_mean::Real)
+  return AcousticPerturbationEquations2D(SVector(v_mean), rho_mean, c_mean)
+end
+
+
+get_name(::AcousticPerturbationEquations2D) = "AcousticPerturbationEquations2D"
+varnames(::typeof(cons2cons), ::AcousticPerturbationEquations2D) = ("v1_prime", "v2_prime", "p_prime")
+varnames(::typeof(cons2prim), ::AcousticPerturbationEquations2D) = ("v1_prime", "v2_prime", "p_prime")
+
+
+"""
+    initial_condition_constant(x, t, equations::AcousticPerturbationEquations2D)
+
+A constant initial condition.
+"""
+function initial_condition_constant(x, t, equations::AcousticPerturbationEquations2D)
+  v1_prime = 0.0
+  v2_prime = 0.0
+  p_prime = 0.0
+
+  return SVector(v1_prime, v2_prime, p_prime)
+end
+
+
+"""
+initial_condition_gauss(x, t, equation::AcousticPerturbationEquations2D)
+
+A Gaussian pulse.
+"""
+function initial_condition_gauss(x, t, equations::AcousticPerturbationEquations2D)
+  v1_prime = 0.0
+  v2_prime = 0.0
+  p_prime = exp(-4*(x[1]^2 + x[2]^2))
+
+  return SVector(v1_prime, v2_prime, p_prime)
+end
+
+
+"""
+    initial_condition_convergence_test(x, t, equations::AcousticPerturbationEquations2D)
+
+A smooth initial condition used for convergence tests in combination with
+[`source_terms_convergence_test`](@ref).
+"""
+function initial_condition_convergence_test(x, t, equations::AcousticPerturbationEquations2D)
+  c = 2.0
+  A = 0.2
+  L = 2.0
+  f = 2.0 / L
+  a = 1.0
+  omega = 2 * pi * f
+  init = c + A * sin(omega * (x[1] + x[2] - a*t))
+
+  v1_prime = init
+  v2_prime = init
+  p = init * init
```

Comment:
```suggestion
  p = init^2
```
That's more clear to me but I don't insist if @sloede feels differently.

Reply:
I agree :+1:

Reply:
I agree, too. Changed it in 284d1ecce8a5ed2158fb767b1613812bf571f5db